common/lib/dependabot/dependency.rb

```diff
@@ -27,17 +27,19 @@ def self.register_display_name_builder(package_manager, name_builder)
     end
 
     attr_reader :name, :version, :requirements, :package_manager,
-                :previous_version, :previous_requirements
+                :previous_version, :previous_requirements, :extra
 
     def initialize(name:, requirements:, package_manager:, version: nil,
-                   previous_version: nil, previous_requirements: nil)
+                   previous_version: nil, previous_requirements: nil,
+                   extra: nil)
       @name = name
       @version = version
       @requirements = requirements.map { |req| symbolize_keys(req) }
       @previous_version = previous_version
       @previous_requirements =
         previous_requirements&.map { |req| symbolize_keys(req) }
       @package_manager = package_manager
+      @extra = extra
```

Comment:
Maybe `metadata` over `extra`?

Reply:
Or `subdependency_metadata` and only allow it for cases where `requirements` is an empty array?

Reply:
Yeah that's nicer 👍